workflow/validate/validate.go

```diff
@@ -1350,14 +1348,20 @@ func verifyNoCycles(tmpl *wfv1.Template, ctx *dagValidationContext) error {
 	return nil
 }
 
-func sortDAGTasks(tmpl *wfv1.Template) error {
+func sortDAGTasks(tmpl *wfv1.Template, ctx *dagValidationContext) error {
 	taskMap := make(map[string]*wfv1.DAGTask, len(tmpl.DAG.Tasks))
 	sortingGraph := make([]*sorting.TopologicalSortingNode, len(tmpl.DAG.Tasks))
 	for index := range tmpl.DAG.Tasks {
-		taskMap[tmpl.DAG.Tasks[index].Name] = &tmpl.DAG.Tasks[index]
+		task := tmpl.DAG.Tasks[index]
+		taskMap[task.Name] = &task
+		dependenciesMap, _ := common.GetTaskDependencies(&task, ctx)
```

Comment:
Will common.GetTaskDependencies return sorted taskname as key?

Reply:
https://stackoverflow.com/questions/18342784/how-to-iterate-through-a-map-in-golang-in-order

Reply:
nope, but do we need an ordered dependencies list here? 🤔

Reply:
My understanding is Dependencies are not in order so validation failed.  Can you test different usecases? like
a->b->c
a-> bc ->d
a->b(has global var),c ->  D depends on (B or C)

Reply:
The reason why validation failed is `sortDAGTasks` only support sort by field `dependencies`, thus some DAG using field `depends` may failed. Here we get task's dependencies list for topological sorting only, and the order after topological sorting is the final order of DAG tasks. So I think the order of `common.GetTaskDependencies` returns is not important here.